app/Http/Controllers/Playlist/PlaylistsController.php

```diff
@@ -564,6 +565,131 @@ public function completeItem(Request $request, $item_id)
         ]);
     }
 
+    /**
+     *
+     * @OA\Get(
+     *     path="/playlists/{playlist_id}/translate",
+     *     tags={"Playlists"},
+     *     summary="Translate a user's playlist",
+     *     operationId="v4_playlists.translate",
+     *     security={{"api_token":{}}},
+     *     @OA\Parameter(
+     *          name="playlist_id",
+     *          in="path",
+     *          required=true,
+     *          @OA\Schema(ref="#/components/schemas/Playlist/properties/id"),
+     *          description="The playlist id"
+     *     ),
+     *     @OA\Parameter(
+     *          name="bible_id",
+     *          in="query",
+     *          required=true,
+     *          @OA\Schema(ref="#/components/schemas/Bible/properties/id"),
+     *          description="The id of the bible that will be used to translate the playlist"
+     *     ),
+     *     @OA\Response(response=200, ref="#/components/responses/playlist")
+     * )
+     *
+     * @param $playlist_id
+     *
+     * @return mixed
+     *
+     *
+     */
+    public function translate(Request $request, $playlist_id)
+    {
+        $user = $request->user();
+
+        // Validate Project / User Connection
+        if (!empty($user) && !$this->compareProjects($user->id, $this->key)) {
+            return $this->setStatusCode(401)->replyWithError(trans('api.projects_users_not_connected'));
+        }
+
+        $bible_id = checkParam('bible_id', true);
+        $bible = Bible::whereId($bible_id)->first();
+
+        if (!$bible) {
+            return $this->setStatusCode(404)->replyWithError('Bible Not Found');
+        }
+
+        $playlist = $this->getPlaylist($user, $playlist_id);
+        if (!$playlist) {
+            return $this->setStatusCode(404)->replyWithError('Playlist Not Found');
+        }
+
+
+        $audio_fileset_types = collect(['audio_drama_stream', 'audio_stream', 'audio_drama', 'audio']);
```

Comment:
if this determines the preferred order of types for the destination, let's change it to audio_stream, audio_drama_stream, audio, audio_drama. (basically prefer streaming over non, and non-music over music)

Reply:
Done